spec/features/admin/dashboard/index_spec.rb

```diff
@@ -0,0 +1,176 @@
+      
+      # Then I see a link to the admin merchants index (/admin/merchants)
+      expect(page).to have_link("Admin Merchant Index")
+      # And I see a link to the admin invoices index (/admin/invoices)
+      expect(page).to have_link("Admin Invoice Index")
+    end
+  end
+  
+  describe "User Story 21" do
+    let!(:merchant1) {create(:merchant)}
+    
+    let!(:item1)     {create(:item, merchant_id: merchant1.id)}
+    
+    let!(:customer1) {create(:customer)} 
+    let!(:customer2) {create(:customer)} 
+    let!(:customer3) {create(:customer)} 
+    let!(:customer4) {create(:customer)} 
+    let!(:customer5) {create(:customer)} 
+    let!(:customer6) {create(:customer)} 
+    
+    let!(:invoice1)  {create(:invoice, customer_id: customer1.id)}
+    let!(:invoice2)  {create(:invoice, customer_id: customer2.id)}
+    let!(:invoice3)  {create(:invoice, customer_id: customer3.id)}
+    let!(:invoice4)  {create(:invoice, customer_id: customer4.id)}
+    let!(:invoice5)  {create(:invoice, customer_id: customer5.id)}
+    let!(:invoice6)  {create(:invoice, customer_id: customer6.id)}
+    
+    let!(:transaction1) {1.times do create(:transaction, invoice_id: invoice1.id, result: 1) end}
+    let!(:transaction2) {2.times do create(:transaction, invoice_id: invoice2.id, result: 1) end}
+    let!(:transaction3) {3.times do create(:transaction, invoice_id: invoice3.id, result: 1) end}
+    let!(:transaction4) {4.times do create(:transaction, invoice_id: invoice4.id, result: 1) end}
+    let!(:transaction5) {5.times do create(:transaction, invoice_id: invoice5.id, result: 1) end}
+    let!(:transaction6) {6.times do create(:transaction, invoice_id: invoice6.id, result: 1) end}
+    # let!(:invoice_item1) {create(:invoice_item,invoice_id: invoice1.id, 
+    #                               item_id: item1.id, quantity: 1, 
+    #                               unit_price: 1, status: 2)}
+    # let!(:invoice_item2) {create(:invoice_item,invoice_id: invoice2.id, 
+    #                               item_id: item1.id, quantity: 2, 
+    #                               unit_price: 1, status: 2)}
+    # let!(:invoice_item3) {create(:invoice_item,invoice_id: invoice3.id, 
+    #                               item_id: item1.id, quantity: 3, 
+    #                               unit_price: 1, status: 2)}
+    # let!(:invoice_item4) {create(:invoice_item,invoice_id: invoice4.id, 
+    #                               item_id: item1.id, quantity: 4, 
+    #                               unit_price: 1, status: 2)}
+    # let!(:invoice_item5) {create(:invoice_item,invoice_id: invoice5.id, 
+    #                               item_id: item1.id, quantity: 5, 
+    #                               unit_price: 1, status: 2)}
+    # let!(:invoice_item6) {create(:invoice_item,invoice_id: invoice6.id, 
+    #                               item_id: item1.id, quantity: 6, 
+    #                               unit_price: 1, status: 2)}
```

Comment:
this huge block I commented out, I left it here in case we need it for the Merchant's Top Five Customers user story. It took a while to write so it would be a shame to type it again ;)

Reply:
Thanks for leaving it in just in case, we can make sure it gets removed with all the rest of the commented out bits later